src/content/ui/adaptive-responsive/layout.md

```diff
@@ -0,0 +1,494 @@
+---
+title: Layout for adaptive apps
+description: >
+  Learn how to create a responsive app >
+  that responds to changes in the screen size.
+short-title: Layout
+---
+
+<?code-excerpt path-base="ui/adaptive_app_demos"?>
+
+<b>TODO(sfshaza): I'm coalescing this page from two previous pages. I'm not done with the text. It likely also needs code or other updates.</b>
+
+Flutter allows you to create apps that self-adapt
+to the device's screen size and orientation.
+
+## General advice
+
+When creating a layout that automatically adapts
+to the size and shape of the user's device,
+there are several things you might keep in mind.
+This page covers how to take an app build for a
+phone and make it beautiful on a wide range of devices.
+
+## Two approaches
+
+There are at least two ways you can create a
+UI that responds to a change in screen size.
+Each approach has its pros and cons.
+
+* `MediaQuery.of`
+  * Returns screen size used by entire app.
+  * Updates layout when user completes a size
+    or orientation change.
+* `LayoutBuilder` 
+  * Returns screen size used by a particular widget.
+  * Updates layout in real time as the user
+    changes size or orientation.
+
+For either approach, you need to determine
+the sizes (called _breakpoints_) where you
+want your UI to change its layout,
+as described in the next section.
+
+### Screen-based breakpoints
+
+The simplest form of procedural layouts uses
+screen-based breakpoints. There are no
+hard and fast rules for the sizes to use
+here, but these are general values:
+
+<?code-excerpt "lib/global/device_size.dart (FormFactor)"?>
+```dart
+class FormFactor {
+  static double desktop = 900;
+  static double tablet = 600;
+  static double handset = 300;
+}
+```
+
+Using breakpoints, you can set up a simple system
+to determine the device type:
+
+<b>TODO(sfshaza): PENDING: Should the following example be updated to use Display instead of MediaQuery?</b>
+
+<?code-excerpt "lib/global/device_size.dart (getFormFactor)"?>
+```dart
+ScreenType getFormFactor(BuildContext context) {
+  // Use .shortestSide to detect device type regardless of orientation
+  double deviceWidth = MediaQuery.of(context).size.shortestSide;
+  if (deviceWidth > FormFactor.desktop) return ScreenType.desktop;
+  if (deviceWidth > FormFactor.tablet) return ScreenType.tablet;
+  if (deviceWidth > FormFactor.handset) return ScreenType.handset;
+  return ScreenType.watch;
+}
+```
+
+As an alternative, you could abstract it more
+and define it in terms of small to large:
+
+<?code-excerpt "lib/global/device_size.dart (ScreenSize)"?>
+```dart
+enum ScreenSize { small, normal, large, extraLarge }
+
+ScreenSize getSize(BuildContext context) {
+  double deviceWidth = MediaQuery.of(context).size.shortestSide;
+  if (deviceWidth > 900) return ScreenSize.extraLarge;
+  if (deviceWidth > 600) return ScreenSize.large;
+  if (deviceWidth > 300) return ScreenSize.normal;
+  return ScreenSize.small;
+}
+```
```

Comment:
I think this section should be removed, and we should just put forward one recommendation that uses the material3 breakpoints referenced earlier. That way we'll be consistent with Material, but also show how to write the code if someone wants to branch away from Material.